packages/nhost-js/src/clients/graphql.ts

```diff
@@ -53,9 +55,10 @@ export class NhostGraphqlClient {
    * @docs https://docs.nhost.io/reference/javascript/nhost-js/graphql/request
    */
   async request<T = any, V = any>(
-    document: string | DocumentNode,
-    variables?: V,
-    config?: AxiosRequestConfig
+    document: string | DocumentNode | TypedDocumentNode<T, V>,
+    ...[variables, config]: V extends never
+      ? [variables?: never, config?: AxiosRequestConfig]
+      : [variables: V, config?: AxiosRequestConfig]
```

Comment:
I have a hard time understanding this code. Can we increase the code readability? Or make a comment explaining what it does?

Reply:
Since this function isn't used named args, and variables was previously optional we need to type it so variables is required when it is passed.

I will add a comment.